javaslang-gwt/src/test/java/client/GwtTestConcurrent.java

```diff
@@ -0,0 +1,56 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package client;
+
+import com.google.gwt.junit.client.GWTTestCase;
+import javaslang.collection.List;
+import javaslang.concurrent.Future;
+import javaslang.concurrent.Promise;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class GwtTestConcurrent extends GWTTestCase {
+
+    @Override public String getModuleName()  {
+        return "TestModule";
+    }
+
+    public void testFutureSuccess() {
+        AtomicBoolean onCompleteCalled = new AtomicBoolean(false);
```

Comment:
There is no reason to use `AtomicBoolean` here in GWT, you can use any boolean holder, for example `boolean[]`.
Also this can help you to remove guava-dependency in `javaslang-gwt`module.
... and fix failed tests :)